### PR TITLE
Clarify configuration handling

### PR DESCRIPTION
**Pull request by timvisee**
[[Rendered]](https://github.com/qdrant/docs/blob/specify-config-loading/qdrant/v1.2.x/configuration.md)

This clarifies how Qdrant handles multiple configuration files. I figured it'd be good to properly document this after <https://github.com/qdrant/qdrant/pull/1972>.

### Tasks
- [x] Write about configuration ordering/priority
- [x] Don't hint to overwrite `production.yaml` by default
- [x] Describe how to use environment variables for config properties
- [x] Describe how a user might read the effective (merged) configuration

### CODE REVIEW

File: qdrant/v1.2.x/configuration.md [hunk 1]
```diff
@@ -3,20 +3,108 @@ title: Configuration
...
+If file 2 or 3 are not found, a warning is shown on startup.
+If file 5 is provided but not found, an error is shown on startup.
+
+Other supported configuration file formats and extensions include: `.toml`, `.json`, `.ini`.
+
+To see the effective (merged) configuration you can query
+[telemetry](https://qdrant.tech/documentation/telemetry/):
+
+```bash
+curl 'http://localhost:6333/telemetry?details_level=6&anonymize=false'
```
Comment (agourlay) [inline qdrant/v1.2.x/configuration.md:66]: this gives the collection configuration but not the server configuration I believe

Comment (timvisee) [reply qdrant/v1.2.x/configuration.md:66]: Correct. I'm seeing the same result.

I think we do have to implement <https://github.com/qdrant/qdrant/issues/1973> then.

**Review by agourlay — COMMENTED** [review]
Thank you for this great addition to our documentation :+1: 

I have only two minor comments to report regarding telemetry.

File: qdrant/v1.2.x/configuration.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (timvisee) [reply qdrant/v1.2.x/configuration.md]: I've removed the telemetry bit for now.

**Review by agourlay — APPROVED** [review]

**Comment by agourlay** [comment]
Reminder to migrate this wonderful PR to the new documentation repository.

**Comment by davidmyriel** [comment]
Concluded via https://github.com/qdrant/landing_page/pull/180

**Pull request closed without merging**